tests/xtimer_usleep_short/main.c

```diff
@@ -22,18 +22,54 @@
 #define TEST_USLEEP_MIN (0)
 #define TEST_USLEEP_MAX (500)
 
+#ifdef BOARD_NATIVE
+/* native can sometime take more time to respond as it is not real time */
+#define TEST_XTIMER_USLEEP_SHORT_SLEEP_MARGIN_US (1000)
+#else
+#define TEST_XTIMER_USLEEP_SHORT_SLEEP_MARGIN_US (20)
+#endif /* BOARD_NATIVE */
+
 int main(void)
 {
     xtimer_sleep(3);
     printf("This test will call xtimer_usleep for values from %d down to %d\n",
-            TEST_USLEEP_MAX, TEST_USLEEP_MIN);
+           TEST_USLEEP_MAX, TEST_USLEEP_MIN);
+
+    printf("Expected delay margin is %d us\n",
+           TEST_XTIMER_USLEEP_SHORT_SLEEP_MARGIN_US);
+
+    uint32_t test_time = 0, sleeping_time = 0, margin_fault = 0;
 
     for (int i = TEST_USLEEP_MAX; i >= TEST_USLEEP_MIN; i--) {
-        printf("going to sleep %d usecs...\n", i);
+        printf("going to sleep %d us\n", i);
+        uint32_t start = xtimer_now_usec();
         xtimer_usleep(i);
+        uint32_t slept = xtimer_now_usec() - start;
+        printf("Slept for      %" PRIu32 " us\n", slept);
+
+        if (slept < (unsigned int)i) {
```

Comment:
as already written several times: I'm against making any checks within in the C-code and would rather do this in the  (python) test script. We should keep the C test code simple avoiding unnecessary maths operations and also increasing the code size with such addition code and output. Its already the case that many tests don't fit on all boards just because of excessive output.

Reply:
Maybe there is a way to get the python script print sequentially?

The idea behind this is to make it obvious if the `XTIMER_BACKOFF` border or something else causes the error.
Thus the print has to be exactly after the sleep, not 5 sleeps after it.

Adding the slept time is neccessary to get a hint of errors like this https://github.com/RIOT-OS/RIOT/pull/9211

Reply:
There are boards that we cannot reboot with `make reset` so cannot run the automated python tests on them. If the test can alone say it is working I find it better.
All the tests based on `embunit` are doing asserts on the node and python only checks for `[SUCCESS]` already.

Also, here they are not `math` operations, they are simple integer comparison and addition.